src/main/java/com/offbytwo/jenkins/model/Build.java

```diff
@@ -38,8 +38,32 @@ public BuildWithDetails details() throws IOException {
         return client.get(url, BuildWithDetails.class);
     }
 	
-	public String Stop() throws HttpResponseException, IOException {
-        return client.get(url + "stop");
+/*
+* This Change (Bad Practice) is due to inconsistencies in Jenkins various versions.
+* Jenkins changed their API from post to get and from get to post and so on.
+* For example version  1.565 is supporting stop method as a post call, version 1.609.1 support it as a get call and version 1.609.2 support it as a post call
+* Thus, when a get is not allowed, an 405 error message is generated "Method Not Allowed" and the stop isn't executed (a post is required).
+* Changed the code in order to do a post call in case a get is not allowed.
+*/
+
+
+    public String Stop() throws HttpResponseException, IOException{
+        try {
+            
+            return client.get(url + "stop");
+        }
+        catch (IOException ex){
+            if(((HttpResponseException)ex).getStatusCode() == 405)
```

Comment:
Jenkins sending an error message: "POST is required for hudson.model.AbstractBuild.doStop
". @pnathan - I'm not sure we will want to address a specific string because, as you can see Jenkins code is not stable - tomorrow it could be a different string.